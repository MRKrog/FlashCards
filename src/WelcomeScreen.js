import React from "react";

import './styles/WelcomeScreen.scss';

const WelcomeScreen = props => {
  if(props.localStorageExist){
    return (
      <section className="WelcomeScreen">
        <article className="welcome-background">
          <div className="welcome-content">
            <span className="welcome-subHeader">Welcome Back</span>
            <h2>Take A Moment To Relax</h2>
            <p>When you're ready, press continue, to finish the quiz.</p>
            <button className="welcome-btn" onClick={props.startQuiz}><i className="fas fa-umbrella-beach"></i> Continue Quiz</button>
          </div>
        </article>
      </section>
    );
  } else {
    return (
      <section className="WelcomeScreen">
        <article className="welcome-background">
          <div className="welcome-content">
            <span className="welcome-subHeader">Welcome To</span>
            <h2>Relaxing <i>Memoize</i> Flash Cards</h2>
            <p>Lets take a brief moment to relax, and once your ready, Press Start (Quiz yourself on HTML, CSS, &amp; Javascript Knowledge).</p>
            <button className="welcome-btn" onClick={props.startQuiz}><i className="fas fa-umbrella-beach"></i> Start Quiz</button>
          </div>
        </article>
      </section>
    );
  }
};

export default WelcomeScreen;
