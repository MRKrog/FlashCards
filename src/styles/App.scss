// font-family: 'Rajdhani', sans-serif;
// font-family: 'Lato', sans-serif;

// Segoe UI", "Roboto", "Oxygen",
  // "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",

$font-title: 'Abel', sans-serif;
$font-header: 'Rajdhani', sans-serif;
$font-copy: 'Lato', sans-serif;

$white: #fff;
$black: #231f20;
$dark-blue: #11212a;

.App {
  background:url(code_bg.jpg);
  width: 100%;
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  z-index: 9;

  &::before {
    content: "";
    position: absolute;
    width: 100%;
    height: 100%;
    background-color: rgba(0, 0, 0, .4);
    background: radial-gradient(ellipse at center, rgba(76,76,76,1) 0%, rgba(17,17,17,0.77) 51%, rgba(19,19,19,0.55) 100%);
  }

  .App-header {
    background-color: #231f20;
    display: block;
    width: 100%;
    z-index: 10;
    box-shadow: 0px 0px 5px 0px rgba(0, 0, 0, .8);

    h1 {
      font-family: $font-title;
      color: #4c4c4c;
      text-transform: uppercase;
      font-size: 2em;
      letter-spacing: .1em;
      text-align: center;
      margin: 0;
      padding: .2em;
      line-height: 1;
      font-weight: 100;
    }

  }

  .main-content {
    width: 100%;
    z-index: 10;

    .welcome-btn {
      width: 100%;
      display: block;
      margin: auto;
      max-width: 200px;
      font-family: "Lato", sans-serif;
      transition: all .5s ease;
      font-size: 1em;
      border: none;
      border-radius: 5px;
      padding: .4em;
      cursor: pointer;
      background: #1f93ec;
      color: #fff;
      outline: none;
    }
  }

  .App-footer {
    background-color: #231f20;
    display: block;
    width: 100%;
    z-index: 10;
    box-shadow: 0px 0px 5px 0px rgba(0, 0, 0, .8);

    p {
      margin: 0;
      text-align: center;
      color: #4c4c4c;
      font-family: $font-title;
      padding: .5em
    }
  }

}
