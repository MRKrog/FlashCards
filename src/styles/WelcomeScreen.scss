$font-header: 'Barlow', sans-serif;
$copyGray: #535667;

.WelcomeScreen {
  max-width: 600px;
  width: 100%;
  display: flex;
  flex-direction: column;
  margin: auto;

  .welcome-background {
    position: relative;
    box-shadow: 0px 1px 7px 1px rgba(35, 31, 32, .2);
    border-radius: 6em;
    background: linear-gradient(135deg, rgba(138,132,10,0.8) 0%, rgba(246,246,246,0.1) 47%, rgba(237,237,237,0.3) 100%);

    &::before {
      content: "";
      position: absolute;
      border-radius: 6em;
      width: 100%;
      height: 100%;
      z-index: 8;
      background: rgba(255,255,255,0.7);
      background: linear-gradient(135deg, rgba(255,255,255,0.7) 0%, rgba(246,246,246,0.65) 47%, rgba(237,237,237,0.83) 100%);
    }

    .welcome-content {
      z-index: 10;
      position: relative;
      padding: 4em 3em 7em 4em;

      .welcome-subHeader {
        font-family: $font-header;
        color: $copyGray;
        line-height: 1;
        font-weight: 400;
      }
      h2 {
        font-family: $font-header;
        font-size: 2em;
        font-weight: 300;
        color: $copyGray;
        line-height: 1;
        padding: .15em 0 .3em 0;
        margin: 0;
      }
      p {
        font-family: $font-header;
        color: $copyGray;
        font-size: 1.1em;
        margin: 0;
        line-height: 1.5;
        font-weight: 400;
      }

      .welcome-btn {
        background: rgba(77, 27, 169, 0.65);
        position: absolute;
        color: rgba(255, 255, 255, .85);
        bottom: 3em;
        right: 4em;
        opacity: 1;
        font-family: $font-header;
        text-transform: uppercase;
        transition: all .3s ease;
        padding: 1em 0;
        border-radius: 2em;
        font-size: .8em;
        outline: none;
        width: 150px;
        border: solid 1px #6a21ab;
        box-shadow: 0px 11px 37px rgba(35, 31, 32, 0.5);

        i {
          padding-right: .5em;
        }

        &:hover {
          opacity: .8;
        }
      }
    }
  }
}
