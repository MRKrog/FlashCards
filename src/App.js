import React, { Component } from 'react';

import Header from './Header.js';
import WelcomeScreen from './WelcomeScreen.js';
import CardContainer from './CardContainer.js';
import Footer from './Footer.js';

import './styles/App.scss';

class App extends Component {
  constructor() {
    super();
    this.state = {
      currentPlayer: false,
      originCards: [],
      allCards: [],
      correctCards: [],
      currCardIndex: 0,
      localStorageExist: false,
      cardsLeft: 0
    }
  }

  componentDidMount() {
    fetch("http://memoize-datasets.herokuapp.com/api/v1/mkCards")
    .then(response => response.json())
    .then(fetchCards => {
      this.setState({
        originCards: fetchCards.mkCards,
        allCards: fetchCards.mkCards
      }, () => { this.getFromLocalStorage() });
    })
    .catch(error => {
      throw new Error(error);
    });
  }

  getFromLocalStorage = () => {
    if(localStorage.length > 0){
      const cardOrigins = this.state.allCards
      const correctArray = JSON.parse(localStorage.getItem('correctQuestions'));
      this.setState({
        correctCards: correctArray,
        localStorageExist: true
      }, () => { this.saveCardOrigins() });
    }

  }

  saveCardOrigins = () => {
    const originalCards = this.state.allCards;
    this.setState({
      originCards: originalCards
    }, () => { this.removeCorrectCards() })

  }

  removeCorrectCards = () => {
    console.log('afe', this.state.originCards);
    const { allCards, correctCards, originCards } = this.state;
    correctCards.forEach((correctCard) => {
      let foundCard = allCards.find(card => card.id === correctCard.id);
      let cardIndex = allCards.indexOf(foundCard);
      allCards.splice(cardIndex, 1);
    });
  }

  handleChangeCard = (clickedAnswer) => {
    const { allCards } = this.state;
    const currentCard = allCards[this.state.currCardIndex];
    let currIndex = this.state.currCardIndex;
    if(currentCard.correctAnswer === clickedAnswer){
      this.state.correctCards.push(currentCard);
      localStorage.setItem('correctQuestions', JSON.stringify(this.state.correctCards));
    }
    currIndex++;
    this.changeCardIndex(currIndex);
  }

  changeCardIndex = (currIndex) => {
    this.setState({
      currCardIndex: currIndex
    });
  }

  startQuiz = () => {
    this.setState({
      currentPlayer: true,
    })
  }

  resetQuiz = () => {
    localStorage.clear();
    this.setState({
      currentPlayer: false,
      localStorageExist: false,
    })

  }

  render() {
    const { allCards, currCardIndex, localStorageExist } = this.state;

    let displayContent;

    if(this.state.currentPlayer === false) {
      displayContent = <WelcomeScreen localStorageExist={localStorageExist}
                                      startQuiz={this.startQuiz} />
    } else {
      displayContent = <CardContainer currentCard={allCards[currCardIndex]}
                                      onChangeCard={this.handleChangeCard} />
    }

    console.log(this.state);
    return (
      <div className="App">
        <Header correctCards={this.state.correctCards}/>
        <main className="main-content">
          {displayContent}
        </main>
        <Footer resetQuiz={this.resetQuiz}
                currentPlayer={this.state.currentPlayer}
              />
      </div>
    )
  }
}

export default App;
