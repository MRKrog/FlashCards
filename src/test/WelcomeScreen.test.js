import React from "react";
import WelcomeScreen from "../WelcomeScreen";
import App from "../App";
import { shallow } from "enzyme";

const startQuiz = jest.fn();
const localStorageExist = App.localStorageExist;

describe("<WelcomeScreen>", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <WelcomeScreen localStorageExist={localStorageExist}
                     startQuiz={startQuiz}
                    />
    );
  });

  it("should match snapshot when all data is passed correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });


});
