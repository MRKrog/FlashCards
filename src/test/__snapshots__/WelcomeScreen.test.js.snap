// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomeScreen> should match snapshot when all data is passed correctly 1`] = `
<section
  className="WelcomeScreen"
>
  <article
    className="welcome-background"
  >
    <div
      className="welcome-content"
    >
      <span
        className="welcome-subHeader"
      >
        Welcome To
      </span>
      <h2>
        Relaxing 
        <i>
          Memoize
        </i>
         Flash Cards
      </h2>
      <p>
        Lets take a brief moment to relax, and once your ready, Press Start (Quiz yourself on HTML, CSS, & Javascript Knowledge).
      </p>
      <button
        className="welcome-btn"
        onClick={[MockFunction]}
      >
        <i
          className="fas fa-umbrella-beach"
        />
         Start Quiz
      </button>
    </div>
  </article>
</section>
`;
